src/AssetControlExtension.php

```diff
@@ -130,21 +130,22 @@ protected function processManipulation(AssetManipulationList $manipulations)
         // When deleting from stage then check if we should archive assets
         $archive = $this->owner->config()->get('keep_archived_assets');
 
-        // Publish assets
-        $this->publishAll($manipulations->getPublicAssets());
-
-        // Protect assets
-        $this->protectAll($manipulations->getProtectedAssets());
-
         // Check deletion policy
         $deletedAssets = $manipulations->getDeletedAssets();
         if ($archive && $this->isVersioned()) {
+            // Publish assets
+            $this->swapAll($manipulations->getPublicAssets());
```

Comment:
Is there a test for this? I can't see any reference of `keep_archived_assets` in tests which would toggle this flag

Reply:
`RedirectKeepArchiveFileControllerTest` test this indirectly. It extends `RedirectFileControllerTest` and runs all the same tests but with `keep_archived_assets` enabled.

`keep_archived_assets` is perhaps a bit counter intuitive in that you can apply it to specific DataObjects without applying it to others. `AssetControlExtensionTest` tests this with `ArchivedObject`.

But to be honest, I worked off the assumption this was already properly tested before I began and didn't look to closely into the existing tests beyond confirming they weren't broken.

Still the `RedirectKeepArchiveFileControllerTest/RedirectFileControllerTest` combo is pretty trough and goes through various scenarios of publishing, unpublishing, archiving, creating a draft, etc.